src/components/renderer/renderer.tsx

```diff
@@ -121,6 +170,19 @@ class Editor extends React.PureComponent<Props, State> {
       .renderer(this.props.renderer)
       .initialize(chart)
       .runAsync();
+
+    // Trigger the resize event if the chart is responsive. This seem to be necessary.
```

Comment:
Let's remove this. Vega should handle this properly, imho. We can leave this for now but file a vega issue and link to it from here.

Reply:
@domoritz, I was about to open an issue in Vega, but with some further investigation, I found that the problem is because we are calling `vegaTooltip.default(view)` before `view.renderer` and `view.initialize`. `vegaTooltip.default` seem to call `view.run` (https://github.com/vega/vega-tooltip/blob/master/src/index.ts), that's why the signals got initialized before we do `view.initialize`

The following code:
```html
<!DOCTYPE html>
<script src="https://cdn.jsdelivr.net/npm/vega@latest"></script>
<script src="https://cdn.jsdelivr.net/npm/vega-tooltip@latest"></script>
<div id="view" style="width: 500px; height: 500px"></div>
<script>
  const spec = {
    $schema: "https://vega.github.io/schema/vega/v5.json",
    signals: [
      {
        name: "width",
        init: "warn(containerSize()[0])",
        on: [{ events: "window:resize", update: "containerSize()[0]" }]
      },
      {
        name: "height",
        init: "warn(containerSize()[1])",
        on: [{ events: "window:resize", update: "containerSize()[1]" }]
      }
    ],
    background: "yellow"
  };

  // editor's initView()
  const view = new vega.View(vega.parse(spec), {
    logLevel: vega.Warn,
    hover: true // enable hover processing
  }).hover();

  console.log("before tooltip");
  vegaTooltip.default(view);
  console.log("after tooltip");

  // editor's renderVega()
  view.renderer("canvas");
  view.initialize(document.getElementById("view"));
  view.runAsync();
</script>
```

I think there is another way to call tooltip. Let me try and I'll send another PR.

Reply:
Nice find. Thanks for looking into this!